File: Server/src/Server.Application/ViewModels/AuthorViewModel.cs
// ReSharper disable UnusedAutoPropertyAccessor.Global
#pragma warning disable CS8618
namespace Server.Application.ViewModels;

public class AuthorViewModel
{
    public Guid Id { get; set; }
    public string FirstName { get; set; }
    public string LastName { get; set; }
    public int BirthYear { get; set; }
    public string ProfilePictureUrl { get; set; }
    public string? Description { get; set; }
}

File: client/src/models/author/AuthorViewModel.ts
import { z } from 'zod';

const AuthorViewModel = z.object({
  id: z.string().uuid(),
  firstName: z.string().max(40),
  lastName: z.string().max(50),
  birthYear: z.number().int().positive(),
  profilePictureUrl: z.string().max(256).nullable(),
  description: z.string().nullable(),
});

export default AuthorViewModel;
export type AuthorViewModelType = z.infer<typeof AuthorViewModel>;
